membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -495,37 +431,12 @@ case class CatalogPlan[+P <: Product, +C <: ChargeList, +S <: Status](
     saving: Option[Int],
     charges: C,
     s: S,
-) extends Plan[P, C] {
-  lazy val slug: String =
-    s"${product.name}-$name".replace(" ", "").toLowerCase
-}
+) extends Plan[P, C]
 
 /** So the benefit of all these type parameters on the higher level models is that you can uniquely identify a particular plan by its type signature
   * and if you can do that then you can pass your super specific (or more generic) plan into subscription service to find the subscription of your
   * dreams
   */
 object SubscriptionPlan {
   type AnyPlan = SubscriptionPlan[Product, ChargeList]
-  type Paid = PaidSubscriptionPlan[Product, PaidChargeList]
-  type Free = FreeSubscriptionPlan[Product, FreeChargeList]
-
-  type Member = SubscriptionPlan[Product.Membership, ChargeList with SingleBenefit[MemberTier]]
-  type PaidMember = PaidSubscriptionPlan[Product.Membership, PaidCharge[Benefit.PaidMemberTier, BillingPeriod]]
-
-  type Supporter = PaidSubscriptionPlan[Product.Membership, PaidCharge[Benefit.Supporter.type, BillingPeriod]]
-  type Partner = PaidSubscriptionPlan[Product.Membership, PaidCharge[Benefit.Partner.type, BillingPeriod]]
-  type Patron = PaidSubscriptionPlan[Product.Membership, PaidCharge[Benefit.Patron.type, BillingPeriod]]
-
-  type ContentSubscription = PaidSubscriptionPlan[Product.ContentSubscription, PaidChargeList]
-  type Digipack = PaidSubscriptionPlan[Product.ZDigipack, PaidCharge[Benefit.Digipack.type, BillingPeriod]]
-  type SupporterPlus = PaidSubscriptionPlan[Product.SupporterPlus, PaidCharge[Benefit.SupporterPlus.type, BillingPeriod]]
-  type Delivery = PaidSubscriptionPlan[Product.Delivery, PaperCharges]
-  type Voucher = PaidSubscriptionPlan[Product.Voucher, PaperCharges]
-  type DigitalVoucher = PaidSubscriptionPlan[Product.DigitalVoucher, PaperCharges]
-
-  type WeeklyZoneA = PaidSubscriptionPlan[Product.WeeklyZoneA, PaidCharge[Weekly.type, BillingPeriod]]
-  type WeeklyZoneB = PaidSubscriptionPlan[Product.WeeklyZoneB, PaidCharge[Weekly.type, BillingPeriod]]
-  type WeeklyPlan = PaidSubscriptionPlan[Product.Weekly, PaidCharge[Weekly.type, BillingPeriod]]
-
-  type Contributor = PaidSubscriptionPlan[Product.Contribution, PaidCharge[Benefit.Contributor.type, BillingPeriod]]
```

Comment:
I made all these unused, which paves the way to remove the type parameter completely from `Subscription` and all the `subscriptionService.current` and other methods throughtout the codebase.